backend/src/main/java/gov/cdc/usds/simplereport/config/GraphQlConfig.java

```diff
@@ -0,0 +1,86 @@
+package gov.cdc.usds.simplereport.config;
+
+import static java.util.Collections.singletonList;
+
+import gov.cdc.usds.simplereport.api.DefaultArgumentValidation;
+import gov.cdc.usds.simplereport.api.directives.RequiredPermissionsDirectiveWiring;
+import gov.cdc.usds.simplereport.api.model.errors.ConflictingUserException;
+import gov.cdc.usds.simplereport.api.model.errors.GenericGraphqlException;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlArgumentException;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlFieldAccessException;
+import gov.cdc.usds.simplereport.api.model.errors.NonexistentUserException;
+import gov.cdc.usds.simplereport.config.scalars.UploadScalarType;
+import gov.cdc.usds.simplereport.config.scalars.datetime.DateTimeScalar;
+import gov.cdc.usds.simplereport.config.scalars.localdate.LocalDateScalar;
+import graphql.GraphQLError;
+import graphql.validation.rules.OnValidationErrorStrategy;
+import graphql.validation.rules.ValidationRules;
+import graphql.validation.schemawiring.ValidationSchemaWiring;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.graphql.execution.DataFetcherExceptionResolver;
+import org.springframework.graphql.execution.RuntimeWiringConfigurer;
+import org.springframework.security.access.AccessDeniedException;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Configuration
+public class GraphQlConfig {
+  public static final String REQUIRED_PERMISSIONS_DIRECTIVE_NAME = "requiredPermissions";
+  public static final int MAXIMUM_SIZE = 256;
+
+  // mask all exception by default
+  // to customize errors returned, you have to manually throw it here
+  @Bean
+  public DataFetcherExceptionResolver dataFetcherExceptionResolver() {
+    return (exception, environment) -> {
+      String errorPath = environment.getExecutionStepInfo().getPath().getSegmentName();
+
+      log.error("Will replace the following exception with a GenericGraphqlException: ", exception);
```

Comment:
non-blocking question: do we need this line for the exception to be logged or will the resolver do it?
mini tiny nit: to me this reads more like an info or debug log message rn - or that it's a problem that we're replacing the exception, which I'm assuming it's not?

Reply:
Its exactly a log entry, since we're wrapping all the exceptions with a generic one, this will make sure that we can still see the actual exception in the logs and inside azureAppInsights